### PR TITLE
Gibson large generation

### PR DESCRIPTION
**Pull request by srama2512**
## Motivation and Context

This PR adds a script to generate customized Gibson training datasets for the PointNav task. Specifically, the Gibson-2+ and Gibson-0+ training splits used in the DD-PPO paper (https://arxiv.org/abs/1911.00357) can be generated by setting the appropriate quality threshold.

## How Has This Been Tested

Tested bug-free execution and the resulting data splits.

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by srama2512** [comment]
Adding this to examples did not seem like a good idea, but I could not think of any other places for adding the script. Please let me know if there's a better organization for this.

File: habitat_baselines/rl/ddppo/data_generation/create_gibson_large_dataset.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        f"{scene_key}.json.gz"
+    )
+    os.makedirs(osp.dirname(out_file), exist_ok=True)
+    with gzip.open(out_file, "wt") as f:
+        f.write(dset.to_json())
+
+
+# Load train / val statistics
+dataset_statistics = json.load(
+    open("./examples/gibson_large_generation/gibson_dset_with_qual.json")
```
Comment (erikwijmans) [inline habitat_baselines/rl/ddppo/data_generation/create_gibson_large_dataset.py:62]:
```suggestion
    open(osp.join(osp.dirname(__file__), "gibson_dset_with_qual.json"), "r")
```

File: habitat_baselines/rl/ddppo/data_generation/create_gibson_large_dataset.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
+"""This script is used to generate customized Gibson training splits for the
```
Comment (mathfac) [inline habitat_baselines/rl/ddppo/data_generation/create_gibson_large_dataset.py:1]: We need to add header:
```
more

#!/usr/bin/env python3

# Copyright (c) Facebook, Inc. and its affiliates.
# This source code is licensed under the MIT license found in the
# LICENSE file in the root directory of this source tree.
```

File: habitat_baselines/rl/ddppo/data_generation/create_gibson_large_dataset.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+import tqdm
+
+import habitat
+import habitat_sim
+from habitat.datasets.pointnav.pointnav_generator import (
+    generate_pointnav_episode,
+)
+
+num_episodes_per_scene = int(1e4)
```
Comment (mathfac) [inline habitat_baselines/rl/ddppo/data_generation/create_gibson_large_dataset.py:22]: Const are upper case per code guidelines:
```suggestion
NUM_EPISODES_PER_SCENE = int(1e4)
```

File: habitat_baselines/rl/ddppo/data_generation/create_gibson_large_dataset.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+print(f"Total number of training scenes: {len(scenes)}")
+
+safe_mkdir("./data/datasets/pointnav/gibson/v2/train_large")
+with multiprocessing.Pool(8) as pool, tqdm.tqdm(total=len(scenes)) as pbar:
+    for _ in pool.imap_unordered(_generate_fn, scenes):
+        pbar.update()
+
+path = "./data/datasets/pointnav/gibson/v2/train_large/train_large.json.gz"
+with gzip.open(path, "wt") as f:
+    json.dump(dict(episodes=[]), f)
```
Comment (mathfac) [inline habitat_baselines/rl/ddppo/data_generation/create_gibson_large_dataset.py:87]: `gibson_dset_with_qual.json` should be mentioned in  [License part of readme](https://github.com/facebookresearch/habitat-api#license) as `Gibson based task datasets and trained models` => `Gibson based task datasets and code generating it, trained models`

File: habitat_baselines/rl/ddppo/data_generation/create_gibson_large_dataset.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        f"./data/datasets/pointnav/gibson/v2/train_large/content/"
+        f"{scene_key}.json.gz"
+    )
+    os.makedirs(osp.dirname(out_file), exist_ok=True)
+    with gzip.open(out_file, "wt") as f:
+        f.write(dset.to_json())
+
+
+# Load train / val statistics
+dataset_statistics = json.load(
```
Comment (mathfac) [inline habitat_baselines/rl/ddppo/data_generation/create_gibson_large_dataset.py:61]: Can we make it as separate function and call it from `main` method. That will make easier to test it and re-use it.

Comment (srama2512) [reply habitat_baselines/rl/ddppo/data_generation/create_gibson_large_dataset.py:61]: Got it. I just need to put the code from hereon into a `generate_gibson_large_dataset()` function and call that from an `if __name_ == "__main__":` block, right?

File: README.md [hunk 6]
```diff
@@ -188,7 +188,7 @@ Habitat-API is MIT licensed. See the [LICENSE file](habitat_baselines/LICENSE) f
 
 The trained models and the task datasets are considered data derived from the correspondent scene datasets.
 - Matterport3D based task datasets and trained models are distributed with [Matterport3D Terms of Use](http://kaldir.vc.in.tum.de/matterport/MP_TOS.pdf) and under [CC BY-NC-SA 3.0 US license](https://creativecommons.org/licenses/by-nc-sa/3.0/us/).
-- Gibson based task datasets and trained models are distributed with [Gibson Terms of Use](https://storage.googleapis.com/gibson_material/Agreement%20GDS%2006-04-18.pdf) and under [CC BY-NC-SA 3.0 US license](https://creativecommons.org/licenses/by-nc-sa/3.0/us/).
+- Gibson based task datasets and code generating it, trained models are distributed with [Gibson Terms of Use](https://storage.googleapis.com/gibson_material/Agreement%20GDS%2006-04-18.pdf) and under [CC BY-NC-SA 3.0 US license](https://creativecommons.org/licenses/by-nc-sa/3.0/us/).
```
Comment (dhruvbatra) [inline README.md:191]:
```suggestion
- Gibson based task datasets, the code for generating such datasets, and trained models are distributed with [Gibson Terms of Use](https://storage.googleapis.com/gibson_material/Agreement%20GDS%2006-04-18.pdf) and under [CC BY-NC-SA 3.0 US license](https://creativecommons.org/licenses/by-nc-sa/3.0/us/).
```

**Comment by erikwijmans** [comment]
Apologies for the slowness on landing this!

**Comment by mathfac** [comment]
Thank you for the contribution!

**Comment by srama2512** [comment]
Glad to be of help!

**Pull request merged**